src/back/wgsl/writer.rs

```diff
@@ -819,56 +877,136 @@ impl<W: Write> Writer<W> {
                 image,
                 sampler,
                 coordinate,
-                array_index: _,
-                offset: _,
+                array_index,
+                offset,
                 level,
-                depth_ref: _,
+                depth_ref,
             } => {
-                // TODO: other texture functions
-                // TODO: comments
-                let fun_name = match level {
-                    crate::SampleLevel::Auto => "textureSample",
-                    _ => {
-                        return Err(Error::Unimplemented(format!(
-                            "expression_imagesample_level {:?}",
-                            level
-                        )));
+                let texture_function = if let Some(depth_ref) = depth_ref {
+                    TextureFunction::SampleCompare(depth_ref)
```

Comment:
is `TextureFunction` actually helping us here? I.e in this if/else/match we could just write the function name, and then later we could grab the expressions out of `level`